ENIGMAsystem/SHELL/Universal_System/image_formats.cpp

```diff
@@ -52,15 +52,17 @@ namespace enigma
 std::map<std::string, ImageLoadFunction> image_load_handlers;
 std::map<std::string, ImageSaveFunction> image_save_handlers;
 
-unsigned long *bgra_to_argb(unsigned char *bgra_data, unsigned pngwidth, unsigned pngheight) {
-  unsigned widfull = nlpo2dc(pngwidth) + 1, hgtfull = nlpo2dc(pngheight) + 1, ih,iw;
+unsigned long *bgra_to_argb(unsigned char *bgra_data, unsigned pngwidth, unsigned pngheight, bool prepend_size) {
+  unsigned widfull = nlpo2dc(pngwidth) + 1, hgtfull = nlpo2dc(pngheight) + 1, ih, iw;
   const int bitmap_size = widfull * hgtfull * 4;
   unsigned char *bitmap = new unsigned char[bitmap_size]();
 
   unsigned i = 0;
-  unsigned elem_numb = pngwidth * pngheight + 2;
+  unsigned elem_numb = pngwidth * pngheight + (prepend_size) ? 2 : 0;
```

Comment:
This will always evaluate to 2 or 0, causing segmentation fault in the following code.

Reply:
either 2 or zero is the whole point, either i want to prepend the two extra values (width and height) or I don't... how would that cause a segmentation fault? Are you sure you looked at the entire code?